### PR TITLE
fix(asr): an ffmpeg the app already bundles is not a missing prerequisite

### PR DESCRIPTION
**Pull request by gambletan**
## Summary

On Apple Silicon — where MLX Whisper is the **default** ASR engine — a clean from-source install fails every transcription with:

```
[Errno 2] No such file or directory: 'ffmpeg'
```

including through the OpenAI-compatible `/v1/audio/transcriptions` route. The docs promise the opposite:

> FFmpeg/FFprobe and yt-dlp are not prerequisites on any install path: the app resolves them itself.

The app *does* resolve them. `imageio-ffmpeg` installs with the backend, its binary is present and runnable, and `find_ffmpeg()` locates it. `MLXWhisperBackend` simply never asked: it handed `mlx_whisper` a path, and given a path `mlx_whisper` calls `whisper.audio.load_audio`, which shells out to a literal `"ffmpeg"` on PATH — the same bare lookup WhisperX was moved off in #479, and one that cannot find the bundled binary because that file is named `ffmpeg-<plat>-vN`.

This was **one call site out of two in the same method**: the forced-alignment step a few lines below already used `_decode_audio_16k_mono`. Decoding once up front and passing the array to both fixes the failure and stops the file being decoded twice per transcription.

## Changes

- `MLXWhisperBackend.transcribe` decodes via the validated ffmpeg once and reuses the array for both `mlx_whisper.transcribe` and `forced_align`
- Regression test asserting `mlx_whisper` is handed decoded audio, not a path
- CHANGELOG entry

## Type

- [x] 🐛 Bug fix

## Testing

Reproduced and verified on a Mac mini M4 Pro (macOS 15, 64 GB), clean from-source install, no system ffmpeg:

**Before** — `POST /v1/audio/transcriptions`, 35.8 s WAV:
```
HTTP 500  {"detail":"[Errno 2] No such file or directory: 'ffmpeg'"}
```

**After** — same request, same machine:
```
HTTP 200  4s cold (model load), 2s warm
{"text":"Hi, I'd like to buy a Chromecast. ..."}
```

The bundled binary was present the whole time:
```
.venv/lib/python3.11/site-packages/imageio_ffmpeg/binaries/ffmpeg-macos-aarch64-v7.1
```

Fail-before / pass-after on the regression test:

```
# on the previous code
FAILED backend/tests/test_asr_mlx_validated_ffmpeg.py::test_mlx_decodes_the_audio_instead_of_handing_over_a_path
E  AssertionError: mlx_whisper was handed a path, so it will resolve ffmpeg itself
   via a bare PATH lookup instead of the validated binary find_ffmpeg() returns

# with this change
1 passed
```

I also wrote a second test asserting transcription survives an empty `PATH`, then deleted it: with `mlx_whisper` mocked it passed both before and after, so it asserted nothing. Not worth carrying.

## Checklist

- Smallest correct change; fixes the class (a decode path that bypasses `find_ffmpeg()`), not just the symptom
- No behaviour change on Windows/Linux — this call site is MLX-only, and MLX is Apple-Silicon-only
- No new network calls; the binary is the one already installed with the backend
- No user-facing strings, so no i18n changes
- CHANGELOG `Unreleased` one-liner added

Happy to adjust the wording or split the double-decode cleanup into its own commit if you'd rather keep the fix minimal.

MLX Whisper now uses the validated bundled `ffmpeg` binary to decode audio once and reuses the waveform for transcription and forced alignment, removing the system `ffmpeg` requirement on Apple Silicon. A regression test verifies that `mlx_whisper.transcribe` receives decoded audio instead of a file path. Reviewers should verify forced alignment behavior with real audio.

### CODE REVIEW

File: backend/tests/test_asr_mlx_validated_ffmpeg.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import sys
+import types
+import wave
+
+import pytest
+
+# conftest.py puts `backend/` on sys.path and points OMNIVOICE_DATA_DIR at a
+# throwaway tmpdir before this module imports the REAL core.config.
+pytest.importorskip("mlx_whisper")
```
[the remark this answers is not in the document]
Comment (gambletan) [reply backend/tests/test_asr_mlx_validated_ffmpeg.py]: Fixed in 46733978. The mocked test no longer skips when mlx_whisper is absent; verified with `.venv/bin/pytest backend/tests/test_asr_mlx_validated_ffmpeg.py -q`.

File: backend/tests/test_asr_mlx_validated_ffmpeg.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    backend = MLXWhisperBackend.__new__(MLXWhisperBackend)
+    backend._model_name = "mlx-community/whisper-large-v3-mlx"
+    backend.transcribe(_silent_wav(tmp_path / "a.wav"), word_timestamps=False)
+
+    audio = seen["audio"]
+    assert not isinstance(audio, (str, bytes)), (
+        "mlx_whisper was handed a path, so it will resolve ffmpeg itself via a "
+        "bare PATH lookup instead of the validated binary find_ffmpeg() returns"
+    )
+    assert hasattr(audio, "__len__") and len(audio) > 0, "decoded audio is empty"
```
[the remark this answers is not in the document]
Comment (gambletan) [reply backend/tests/test_asr_mlx_validated_ffmpeg.py]: Fixed in 46733978. The fixture now returns a segment, enables word timestamps, and asserts forced_align receives the identical decoded waveform object; the focused test passes.
